### PR TITLE
feat: silent programmatic usage

### PR DESCRIPTION
**Pull request by millsp**
### Description

At Prisma, we were looking for a robust and nice abstraction to handle multiple package manager installs/commands. We've found you :smile: ! We plan to use some of `ni`'s internals in order to reliably run package manager commands via our own CLI. We are currently using `getCommand` for that. However, we don't want `ni`'s output to leak warnings as we would like to handle that ourselves.

For this reason, I implemented a silent/programmatic mode which, right now, would be available for programmatic usages. Maybe later on it would be the base to bring a `--silent` for instance, but for now is limited to our own use-case. It prevents most console logging except for `--help` and `debug`, while also preventing prompts, and throwing errors instead of `process.exit`.

### CODE REVIEW

File: src/fs.ts [hunk 1]
```diff
@@ -11,7 +12,7 @@ export function getPackageJSON(cwd = process.cwd()): any {
       return data
     }
     catch (e) {
-      console.warn('Failed to parse package.json')
+      !options.silent ? console.warn('Failed to parse package.json') : 0
       process.exit(0)
```
Comment (millsp) [inline src/fs.ts:16]: Unrelated to this PR, but I wondered is the exit code was intentionally made `0` instead of `1`?

File: src/fs.ts [hunk 2]
```diff
@@ -11,8 +12,9 @@ export function getPackageJSON(cwd = process.cwd()): any {
       return data
     }
     catch (e) {
-      console.warn('Failed to parse package.json')
-      process.exit(0)
+      !programmatic ? console.warn('Failed to parse package.json') : 0
+      !programmatic ? process.exit(0) : 0
```
Comment (millsp) [inline src/fs.ts:16]: Is was wondering if `exit 0` was intentional here?

Comment (antfu) [reply src/fs.ts:16]: Yeah probably not. 

Can we do an `if` instead of ternary conditions?

File: src/commands/nr.ts [hunk 3]
```diff
@@ -12,15 +12,16 @@ runCli(async (agent, args, ctx) => {
 
   if (args[0] === '-') {
     if (!storage.lastRunCommand) {
-      console.error('No last command found')
-      process.exit(1)
+      !ctx.programmatic ? console.error('No last command found') : 0
+      !ctx.programmatic ? process.exit(1) : 0
```
Comment (antfu) [inline src/commands/nr.ts:16]: Same as here

File: src/fs.ts (hunk 2 above)
Comment (millsp) [reply src/fs.ts]: Sure, I also fixed some other linting issues and adding some tests as well.

Comment (millsp) [reply src/fs.ts]: (Also changed exit code)

**Pull request merged**